keanu-project/src/main/java/io/improbable/keanu/algorithms/NetworkSamples.java

```diff
@@ -78,8 +82,9 @@ public NetworkSamples downSample(final int downSampleInterval) {
                 e -> downSample(e.getValue(), downSampleInterval)
                 )
             );
+        final List<Double> withLogProbsDownSampled = (List<Double>) downSample(logOfMasterP, downSampleInterval);
 
-        return new NetworkSamples(withSamplesDownSampled, size / downSampleInterval);
+        return new NetworkSamples(withSamplesDownSampled, withLogProbsDownSampled, size / downSampleInterval);
```

Comment:
This just seems to be the pattern this code uses at the moment but obviously for large network samples recreating all this stuff is going to have pretty high transitory memory usage and is not gonna be great performance wise.  Should we be considering "inPlace" kind of operations?